src/cryptoadvance/specter/notifications/notification_manager.py

```diff
@@ -0,0 +1,367 @@
+              └───────────────────────┘ │                       └───────────────────────┘
+                                        │                       ┌───────────────────────┐
+                                        └──────────────────────►│   JSNotifications     │
+                                                                └───────────────────────┘
+
+    2. internal_notification  at the example of on_close
+
+                  ┌───────────────────────┐
+                  │                       │  if "internal_notification" in notification.target_uis:
+                  │  NotificationManager  │ ──────────────────────────────────────►┐
+                  │   .create_and_show    │                                        │
+                  └───────────────────────┘                                        ▼
+                                                                 ┌────────────────────────────┐
+                                                                 │  NotificationManager       │
+                                                                 │   .treat_internal_message  │
+                                                                 │     e.g. on_close          │
+                                                                 └─────────────────┬──────────┘
+                                                                                   │ on_close
+                                                                                   ▼
+                   ┌───────────────────────┐                         ┌───────────────────────┐
+                   │   JSNotifications     │ ◄─────────────────────┐ │   NotificationManager │
+                   │   .on_close           │                       │ │   ._internal_on_close │
+                   └───────────────────────┘                       │ └───────────────────────┘
+                   ┌───────────────────────┐                       │
+                   │  NotificationManager  │                       │
+                   │  .delete_notification │ ◄─────────────────────┘
+                   └───────────────────────┘
+
+    """
+
+    def __init__(self, websockets_port, user_manager, ui_notifications=None):
+        """
+        Arguments:
+            - ui_notifications:  {user_id: [list of ui_notifications]}
+                    The "default" ui_notifications is at position 0
+        """
+        self.ui_notifications = ui_notifications if ui_notifications else []
+        self.notifications = []
+        self.register_default_ui_notifications()
+
+        (
+            self.websockets_server,
+            self.websockets_client,
+        ) = websockets_server_client.create_websockets_server_and_client(
+            websockets_port, user_manager, self
+        )
+        websockets_server_client.run_websockets_server_and_client(
+            self.websockets_server, self.websockets_client
+        )
+
+    def find_target_ui(self, target_ui, user_id):
```

Comment:
This method is used only internally. Can you mark all methods which are used internally only?
```suggestion
    def _find_target_ui(self, target_ui, user_id):
```
Maybe you can also reshuffle the official API-methods to the top? This would greatly improve the understanding. I had projects (in DotNet) where i did nothing else for weeks than putting methods private (or refactor stuff to make more things private) in order to better understand the overall architecture and make changes understanding what the impact of such changes would be.

Reply:
sure. No problem.